Watch App

algesten opened the pull request:
Need to update submodules again once we merge https://github.com/mattrubin/Base32/pull/1 and https://github.com/mattrubin/OneTimePassword/pull/94

algesten committed:
ignore .DS_Store

algesten committed:
Watch App

algesten commented:
I don't even understand what travis is complaining about :/

beaucollins commented:
I think travis is missing the watch simulator so it's not able to run build the project.

```
Testing failed:
	Target specifies product type 'com.apple.product-type.watchkit2-extension', but there's no such product type for the 'iphonesimulator' platform
	Unable to run command 'PBXCp Base32.framework' - this target might include its own product.
	Unable to run command 'PBXCp OneTimePassword.framework' - this target might include its own product.
** TEST FAILED **
```

algesten commented:
wonder whether it can be that the project settings say "latest watchos 3.1" while the travis conf says

```
matrix:
  include:
    - osx_image: xcode7.3
      xcode_sdk: iphonesimulator9.3
```

watchos 3 is only iOS 10, so it may be that a 9.3 simulator can never run this.

algesten commented:
but then to contradict myself – i think i built everything to be watchos 9 compatible, only using watchos 2 API since the rest of the project is thereabouts.

beaucollins commented on Authenticator.xcodeproj/project.pbxproj:
```diff
@@ -640,13 +959,216 @@
+				SWIFT_OPTIMIZATION_LEVEL = "-Onone";
+				SWIFT_VERSION = 3.0;
+				TARGETED_DEVICE_FAMILY = 4;
+				WATCHOS_DEPLOYMENT_TARGET = 3.0;
```
This might be why travis can't build it, try with 2.0.

algesten replied:
cool! i give it a go

algesten replied:
still no luck.

```
	Target specifies product type 'com.apple.product-type.watchkit2-extension', but there's no such product type for the 'iphonesimulator' platform
```

the error message is a bit curious. it's like it's trying to run the watch extension on a iphone simulator, which won't work.

mattrubin committed:
Update dependencies

mattrubin committed:
Add keychain reset function

From https://github.com/mattrubin/OneTimePassword/pull/94

mattrubin committed:
Add Token toURLWithSecret()

See https://github.com/mattrubin/OneTimePassword/pull/94

algesten commented:
I've rebased and removed unnecessary commits. Do you want to the keychain reset and toURLWithSecret() as separate commits, or should I squash it to one "watch app" commit?
